framework/src/base/ExecFlagRegistry.C

```diff
@@ -0,0 +1,37 @@
+#include "ExecFlagRegistry.h"
+
+#include "MooseUtils.h"
+
+namespace moose
+{
+namespace internal
+{
+
+ExecFlagRegistry &
+getExecFlagRegistry()
+{
+  static ExecFlagRegistry exec_flag_registry;
+  return exec_flag_registry;
+}
+
+ExecFlagRegistry::ExecFlagRegistry()
+  : GenericRegistry<std::string, MooseEnumItem>("ExecFlagRegistry")
+{
+}
+
+const MooseEnumItem &
+ExecFlagRegistry::registerFlag(const std::string & name)
+{
+  const auto create = [&name](const std::size_t id) { return MooseEnumItem(name, id); };
+  const auto id = registerItem(MooseUtils::toUpper(name), create);
+  return item(id);
+}
+
+bool
+ExecFlagRegistry::isFlagRegistered(const MooseEnumItem & flag) const
+{
+  return keyExists(flag.name()) && item(id(flag.name())).id() == flag.id();
```

Comment:
Is `item(id(flag.name())).id()` not guaranteed to equal `id(flag.name())`?

Reply:
Yeah - that's the whole point of the registry. Someone could go register `EXEC_FOO` with an ID (assigned by this) as 5. But then I could do randomly create a `MooseEnumItem` with name `EXEC_FOO` and id 6, in which case I wouldn't want _that_ item to be registered.

Reply:
Yeah - that's the whole point of the registry. Someone could go register `EXEC_FOO` with an ID (assigned by this) as 5. But then I could do randomly create a `MooseEnumItem` with name `EXEC_FOO` and id 6, in which case I wouldn't want _that_ item to be registered.

Reply:
Kk